python/cuml/manifold/umap.pyx

```diff
@@ -738,6 +772,9 @@ class UMAP(UniversalBase,
             cdef UMAPParams* umap_params = \
                 <UMAPParams*> <size_t> UMAP._build_umap_params(self,
                                                                self.sparse_fit)
+            # NN Descent doesn't support transform yet
+            umap_params.build_algo = graph_build_algo.BRUTE_FORCE_KNN
```

Comment:
Would be important to log this users. It looks like this is switched silently from Python side and could then prevent the C++ side from hitting the exception.